File: Hackerrank/Algorithms/Implementation/kangaroo/kangaroo.php
<?php
/*
   Link - https://www.hackerrank.com/challenges/kangaroo/problem
*/
// Complete the kangaroo function below.
function kangaroo($x1, $v1, $x2, $v2) {

if(($v2-$v1) == 0){
    return "NO";
}
   if(($x1-$x2)%($v2 - $v1) == 0 and ($x1-$x2)/($v2 - $v1)>0){
        return "YES";
    }else{
        return "NO";
    }
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$stdin = fopen("php://stdin", "r");

fscanf($stdin, "%[^\n]", $x1V1X2V2_temp);
$x1V1X2V2 = explode(' ', $x1V1X2V2_temp);

$x1 = intval($x1V1X2V2[0]);

$v1 = intval($x1V1X2V2[1]);

$x2 = intval($x1V1X2V2[2]);

$v2 = intval($x1V1X2V2[3]);

$result = kangaroo($x1, $v1, $x2, $v2);

fwrite($fptr, $result . "\n");

fclose($stdin);
fclose($fptr);

File: Hackerrank/kangaroo.cpp
#include<stdio.h>
void kangaroo(int x1,int v1,int x2,int v2)
{
    int l,i,m1,m2,c=1;
    for(i=0;i<10000;i++)
    {
        m1=x1+(v1*c);
        m2=x2+(v2*c);
        if(m1==m2)
        {printf("YES");
         break;}
        c+=1;
    }
    if(m1!=m2)
        printf("NO");
    
    
    }

int main()
{
    int x1,v1,x2,v2,l,k;
    scanf("%d%d%d%d",&x1,&v1,&x2,&v2);
    kangaroo(x1,v1,x2,v2);
    return 0;
}
